extensions/XeroName/Deltatime.js

```diff
@@ -0,0 +1,223 @@
+    fT += (thisFT - fT) / filtStren;
+
+    rawFPS = 1000 / fT;
+
+    if (rawFPS < 0.5) {
+      vmFPS = rawFPS.toFixed(2);
+    } else {
+      vmFPS = Math.round(rawFPS);
+    }
+
+    if (vmFPS > 0) {
+      vmDt = 1 / vmFPS; // At normal situation
+    } else {
+      vmDt = 0; // Prevent situation of "DIV/0"
+    }
+    last = now;
+  };
+
+  class Dt {
+    getInfo() {
+      return {
+        id: 'dtbyxeroname',
+        name: 'Deltatime',
+        docsURI: 'https://xeroname.github.io/dtbyxeroname/',
+        color1: '#333333',
+        color2: '#444444',
+        color3: '#ffffff',
+        menuIconURI: icon,
+        blocks: [
+          {
+            opcode: 'getDt_vm',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'ΔT'
+          },
+          {
+            opcode: 'getFPS_vm',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'FPS'
+          },
+          {
+            opcode: 'getTimeRes',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'Get [TYPE] of [TARGET]',
+            arguments: {
+              TYPE: {
+                type: Scratch.ArgumentType.STRING,
+                menu: 'type_menu'
+              },
+
+              TARGET: {
+                type: Scratch.ArgumentType.STRING,
+                menu: 'target_menu'
+              }
+            },
+            hideFromPalette: true
+          },
+          {
+            opcode: 'isFPSposi',
+            blockType: Scratch.BlockType.BOOLEAN,
+            text: 'FPS > 0',
+            hideFromPalette: true
+          },
+          {
+            blockType: "label",
+            text: "Filtering"
+          },
+          {
+            opcode: 'setFilterStrength',
+            blockType: Scratch.BlockType.COMMAND,
+            text: 'Set Filter Strength : [STRENGTH]',
+            arguments: {
+              STRENGTH: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '8'
+              }
+            }
+          },
+          {
+            opcode: 'getFilterStrength',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'Filter Strength'
+          },
+          {
+            blockType: "label",
+            text: "Calculating"
+          },
+          {
+            opcode: 'setCalculatorStandard',
+            blockType: Scratch.BlockType.COMMAND,
+            text: 'Set Calculator Standard to [FPS] FPS',
+            arguments: {
+              FPS: {
+                type: Scratch.ArgumentType.NUMBER,
+                defaultValue: '60'
+              }
+            }
+          },
+          {
+            opcode: 'getCalculatorStandard',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'Standard FPS of Calculator',
+            disableMonitor: true
+          },
+          {
+            opcode: 'calcMultiplyValue',
+            blockType: Scratch.BlockType.REPORTER,
+            text: 'Calc [DISTANCE]',
```

Comment:
is there a better way to name these?

Reply:
Now that block will be named as "**Value to Move [DISTANCE] with ΔT**".